base/doc/clsguide.tex

```diff
@@ -1181,12 +1181,162 @@ \subsection{Querying localisation}
 where |#1| is the user input and the first argument to
 \cs{MakeUppercaseAux} takes two arguments, the locale and input text.
 
+\subsection{Extended and expandable references of properties}
+
+A property is something that \LaTeX{} can track, such as a page
+number, a section number or a name. Such properties can be labeled,
+their current value written to the \texttt{aux}-file and referenced in
+the next compilation, similar to the standard \cs{label}/\cs{ref}
+commands.
+
+\begin{decl}
+  |\RecordProperties|\arg{label}\arg{list of properties} 
+\end{decl}
+This command writes the current or shipout value (depending on the
+declaration of the property) of the properties in the list into the
+\texttt{aux}-file labeled by \meta{label}. The arguments can contain
+commands which are expanded. \meta{label} can expand to an arbitrary
+string (as long as it can safely be written to the \texttt{aux}-file)
+but note that the label names of \cs{label} and \cs{RecordProperties}
+share a singe namespace. That means that you get a \texttt{Label `A'
+  multiply defined} warning with the following code:
+\begin{verbatim}
+\label{A}\RecordProperties{A}{page}
+\end{verbatim}
+
+
+\begin{decl}
+ |\RefProperty|\arg{label}\arg{property}
+\end{decl}
+This command allows to reference the value of the property
+\meta{property} labeled by \meta{label}. Differently to the standard
+\cs{ref} command the command is expandable and the value can for
+example---if it is a number---be used in an assignment.
+
+\begin{verbatim}
+\section{A section}
+\RecordProperties{mylabel}{pagenum,counter}
+\RefProperty{mylabel}{counter} % outputs section
+\setcounter{mycounter}{\RefProperty{mylabel}{pagenum}} 
+\end{verbatim}
+
+
+
+As \cs{RefProperty} is expandable it can not issue a rerun warning if
+a label is not found. If needed such a warning can be forced by the
+following command:
+\begin{decl}
+ |\RefUndefinedWarn|\arg{label}\arg{property}
+\end{decl}
+
+\LaTeX{} predefines a set of properties, this set contains also the
+properties stored by the standard \cs{label} command:
+\begin{description}
+ \item[\texttt{abspage}] The absolute value of the current page:
+   starts at $1$ and increases monotonically at each shipout.
+   
+ \item[page] The current page as given by \cs{thepage}: this may or
+   may not be a numerical value, depending on the current
+   style. Contrast with \texttt{abspage}. You get this value also with
+   the standard \cs{label}/\cs{pageref}.
+
+ \item[\texttt{pagenum}] The current page as arabic number. This is
+   suitable for integer operations and comparisions.
+
+ \item[\texttt{currentlabel}] The content of \cs{@currentlabel}. This
+   is the value that you get also with the standard
+   \cs{label}/\cs{ref}.
+
+ \item[\texttt{title}] The content of \cs{@currentlabelname}.  This
+   command is filled beside others by the \pkg{nameref} package and
+   some classes (e.g.~\pkg{memoir}) and typically gives the title of
+   some sectioning command.
```

Comment:
Maybe I’m just tired but I first read this as meaning it would contain a string like “Chapter”. Perhaps “typically gives the title defined in the document by some sectioning command”